### PR TITLE
License project and add a REUSE compliance check workflow to the CI

### LINKED ISSUE #16: Properly licence all work in this repo and add REUSE linter in CI
And perhaps reuse badge in readme.

### PR DESCRIPTION
**Pull request by vascoguita**
Closes #16

### CODE REVIEW

**Comment by gnulabis** [comment]
Ok, 3 comments:

1. Copyright for work paid by CERN is not personal, it belongs to CERN. So "2023 CERN (home.cern)".
2. For images and other binary files it is better to use the `.reuse` structure instead of having one license file per image. See for example: https://gitlab.cern.ch/be-cem-edl/fec/hardware-modules/wrtd-reference-designs/-/blob/master/.reuse/dep5
3. For CreativeCommons it should be "CC-BY-SA-4.0+" (note the plus sign) to denote "4 or later".

**Comment by vascoguita** [comment]
>     1. Copyright for work paid by CERN is not personal, it belongs to CERN. So "2023 CERN (home.cern)".

Yes of course, I'll change it.
But I need an email address to comply with SPDX:

```
# SPDX-FileCopyrightText: [year] [copyright holder] <[email address]>
#
# SPDX-License-Identifier: [identifier]
```
 
>     2. For images and other binary files it is better to use the `.reuse` structure instead of having one license file per image. See for example: https://gitlab.cern.ch/be-cem-edl/fec/hardware-modules/wrtd-reference-designs/-/blob/master/.reuse/dep5

Alright, I'm working on that.
 
>     3. For CreativeCommons it should be "CC-BY-SA-4.0+" (note the plus sign) to denote "4 or later".


According to the REUSE tool CC-BY-SA-4.0+ is not a valid SPDX identifier:
```
'CC-BY-SA-4.0+' is not a valid SPDX License Identifier.

Did you mean:
* CC-BY-4.0
* CC-BY-NC-4.0
* CC-BY-NC-SA-4.0
* CC-BY-ND-4.0
* CC-BY-SA-1.0
* CC-BY-SA-2.0
* CC-BY-SA-2.0-UK
* CC-BY-SA-2.1-JP
* CC-BY-SA-2.5
* CC-BY-SA-3.0
* CC-BY-SA-3.0-AT
* CC-BY-SA-3.0-DE
* CC-BY-SA-4.0
```

**Comment by gnulabis** [comment]
We had already looked at the email issue with SPDX and we had concluded that for companies / organisations it's ok to put a website url instead. That's how we arrived to (home.cern) and we use it everywhere.

**Comment by vascoguita** [comment]
> We had already looked at the email issue with SPDX and we had concluded that for companies / organisations it's ok to put a website url instead. That's how we arrived to (home.cern) and we use it everywhere.

Understood, it's fixed now.

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Upstream-Contact: CERN (home.cern)
+Source: https://github.com/OHWR/ohwr.org
+
+Files: static/images/*
+Copyright: 2023 CERN (home.cern)
+License: CC-BY-SA-4.0+
+
+Files: go.sum
+Copyright: 2023 CERN (home.cern)
+License: BSD-3-Clause
```
Comment (gnulabis) [inline .reuse/dep5:12]: I think you can remove this file from the repo and gitignore this file. I did it as a test and running `hugo` regenerates it.

Comment (vascoguita) [reply .reuse/dep5:12]: yes, go.sum is generated automatically from go.mod but it's one of those rare cases where it's a good practice to commit an automatically generated file.

https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file

Comment (gnulabis) [reply .reuse/dep5:12]: Awesome, thanks for the explanation.

**Comment by gnulabis** [comment]
Fine with me, you can merge.

(is there a way to approve like in gitlab?)

**Review by gnulabis — APPROVED** [review]

**Comment by vascoguita** [comment]
In GitHub one can review/approve a PR on the **Files changed** tab:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews

**Pull request merged**